zio-http/src/main/scala/zio/http/ServerConfig.scala

```diff
@@ -0,0 +1,182 @@
+                           channelType: ChannelType = ChannelType.AUTO,
+                           nThreads: Int = 0,
+                         ) {
+    self =>
+    def useAggregator: Boolean = objectAggregator >= 0
+
+    /**
+     * Creates a new server using a HttpServerExpectContinueHandler to send a
+     * 100 HttpResponse if necessary.
+     */
+    def withAcceptContinue(enable: Boolean): Config = self.copy(acceptContinue = enable)
+
+    /**
+     * Creates a new server listening on the provided hostname and port.
+     */
+    def withBinding(hostname: String, port: Int): Config = self.copy(address = new InetSocketAddress(hostname, port))
+
+    /**
+     * Creates a new server listening on the provided InetAddress and port.
+     */
+    def withBinding(address: InetAddress, port: Int): Config =
+      self.copy(address = new InetSocketAddress(address, port))
+
+    /**
+     * Creates a new server listening on the provided InetSocketAddress.
+     */
+    def withBinding(inetSocketAddress: InetSocketAddress): Config = self.copy(address = inetSocketAddress)
+
+    /**
+     * Creates a new server with FlushConsolidationHandler to control the
+     * flush operations in a more efficient way if enabled (@see <a
+     * href="https://netty.io/4.1/api/io/netty/handler/flush/FlushConsolidationHandler.html">FlushConsolidationHandler<a>).
+     */
+    def withConsolidateFlush(enable: Boolean): Config = self.copy(consolidateFlush = enable)
+
+    /**
+     * Creates a new server using netty FlowControlHandler if enable (@see <a
+     * href="https://netty.io/4.1/api/io/netty/handler/flow/FlowControlHandler.html">FlowControlHandler</a>).
+     */
+    def withFlowControl(enable: Boolean): Config = self.copy(flowControl = enable)
+
+    /**
+     * Creates a new server with netty's HttpServerKeepAliveHandler to close
+     * persistent connections when enable is true (@see <a
+     * href="https://netty.io/4.1/api/io/netty/handler/codec/http/HttpServerKeepAliveHandler.html">HttpServerKeepAliveHandler</a>).
+     */
+    def withKeepAlive(enable: Boolean): Config = self.copy(keepAlive = enable)
+
+    /**
+     * Creates a new server with the leak detection level provided (@see <a
+     * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html">ResourceLeakDetector.Level</a>).
+     */
+    def withLeakDetection(level: LeakDetectionLevel): Config = self.copy(leakDetectionLevel = level)
+
+    /**
+     * Creates a new server with HttpObjectAggregator with the specified max
+     * size of the aggregated content.
+     */
+    def withObjectAggregator(maxRequestSize: Int = 1024 * 100): Config =
+      self.copy(objectAggregator = maxRequestSize)
+
+    /**
+     * Creates a new server listening on the provided port.
+     */
+    def withPort(port: Int): Config = self.copy(address = new InetSocketAddress(port))
+
+    /**
+     * Creates a new server with netty's HttpContentDecompressor to decompress
```

Comment:
These Scaladoc can be changed to reflect the fact that none of these methods create a server, they just modify the configuration.